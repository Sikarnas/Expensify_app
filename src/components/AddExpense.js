import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startAddExpense} from '../actions/expenses';

export class AddExpense extends React.Component { 
    onSubmit = (expense) => {
        // props.dispatch(addExpense(expense))
        this.props.startAddExpense(expense)
        this.props.history.push('/dashboard')
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="ui container">
                        <h1 className="page-header__title">Add expense</h1>        
                    </div>
                </div>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
             </div> 
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startAddExpense: (expense) => dispatch(startAddExpense(expense))
    }
}

export default connect(undefined, mapDispatchToProps)(AddExpense);