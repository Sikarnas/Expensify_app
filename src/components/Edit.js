import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from '../actions/expenses'

export class Edit extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id,expense)
        this.props.history.push('/dashboard')
    }
    onClick = () => {
        this.props.startRemoveExpense(this.props.expense.id)
            this.props.history.push('/dashboard')
     }
     
    render() {
        return (
            <div className="ui container">
             <h1 className="page-header__title">Edit expense</h1>        
                <div className="edit">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <div className="remove">
                        <button onClick={this.onClick} className="ui red button">Remove</button>
                    </div>
                </div>
                
            </div>
         )
    }   
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id,expense)),
        startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(Edit)