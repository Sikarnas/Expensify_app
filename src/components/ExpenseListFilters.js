import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters'
import {DateRangePicker} from 'react-dates'

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}))
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSelectChange = (e) => {
        e.target.value === 'date' 
        ? this.props.sortByDate()
        : this.props.sortByAmount()
    }
    render() {
        return (
            <div className="ui container">
            <div className="input-group">
                <div className="input_group__item ">
                    <input className="text-input" 
                    type='text' 
                    value={this.props.filters.text} 
                    onChange={this.onTextChange}
                    placeholder="Search expenses"
                    />
                </div>
                <div className="input_group__item">
                        <select 
                            className="select"
                            value = {this.props.filters.sortBy} 
                            onChange={this.onSelectChange}
                            >
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                </div>
                <div className="input_group__item">
                    <DateRangePicker
                        startDateId='1'
                        startDate={this.props.filters.startDate}
                        endDate={this.props.filters.endDate}
                        endDateId='365'
                        onDatesChange={this.onDatesChange}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                </div>
            </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
        filters: state.filters
})

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);