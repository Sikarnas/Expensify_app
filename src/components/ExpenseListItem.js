import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = (props) => (
// const ExpenseListItem = ({description, createdAt, amount, id}) => (
        <Link className="list-item" to={`/edit/${props.id}`}>
            <div>
                <h3 className="list_item__title">{props.description}</h3>
                <span className="list_item_subtitle">{moment(props.createdAt).format('YYYY-MM-DD')}</span>
            </div>
            <div>
                <h3 className="list-item__amount">{numeral(props.amount/100).format('00.00')} â‚¬</h3>
            </div>
        </Link>
)

export default (ExpenseListItem);
