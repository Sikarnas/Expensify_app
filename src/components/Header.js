import React from 'react';
import { Link } from 'react-router-dom';
import {startLogout} from '../actions/auth'
import {connect} from 'react-redux'

export const Header = ({startLogout}) => (
        <header className="header">
        <div className="ui container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard" exact={'true'}><h1><span/> <i className="money bill alternate outline icon"></i>Expense App</h1></Link>
                {/* <NavLink to="/help" activeClassName="is-active">Help</NavLink> */}
                <button className="ui inverted button" onClick={startLogout}><i className="user outline icon"></i>Logout</button>
            </div>
        </div>   
    </header>

    
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)

