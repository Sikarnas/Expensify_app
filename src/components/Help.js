import React from 'react';

const Help = (props) => (
    <p>HALLLP PLES</p>
);

export default Help