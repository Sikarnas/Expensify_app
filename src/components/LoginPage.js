import React from 'react';
import {connect} from 'react-redux'
import {startLogin} from '../actions/auth'

const LoginPage = (props) => (
    <div className="login_box">
        <div className="login_box__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>Track your expenses with ease</p>
            <button className="ui primary button" onClick={props.startLogin}><i className="google icon"></i> Login with Google</button>
        </div>   
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)