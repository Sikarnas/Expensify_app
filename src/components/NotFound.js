import React from 'react';
import {Link} from 'react-router-dom';

const NotFound = (props) => (
    <p>404 SEND HELP - go back to <Link to="/">home</Link></p>
);

export default NotFound