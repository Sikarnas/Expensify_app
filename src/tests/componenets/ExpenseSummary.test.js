import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary'

test('should correctly render expensesSummary with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={512}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render expensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={11} expensesTotal={5544512}/>)
    expect(wrapper).toMatchSnapshot()
})