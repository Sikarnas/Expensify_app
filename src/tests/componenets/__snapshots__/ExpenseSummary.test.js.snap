// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    11
     
    expenses
     totaling 
    55445.12
  </h1>
</div>
`;

exports[`should correctly render expensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    05.12
  </h1>
</div>
`;
